opentelemetry-sdk/src/opentelemetry/sdk/propagator/b3_format.py

```diff
@@ -92,7 +93,11 @@ def extract(cls, get_from_carrier, carrier):
         if sampled in cls._SAMPLE_PROPAGATE_VALUES or flags == "1":
             options |= trace.TraceOptions.RECORDED
 
-        return trace.SpanContext(
+        # TODO: there is no standard way to retrieve
```

Comment:
The actual context used to be an implementation detail. You are making it a public interface now. When we want to switch to using just plain `contextvars` in the future, that may be a problem.

Reply:
I don't think contextvars really solves this, as you still need some sort of public interface to define where to retrieve or set context-related values. 

I agree that's what's happening here. Maybe @reyang has opinions on this? I'm +1 for just being explicit with what slots should be populated with which objects.

Reply:
Other option is another interface which handles specifying that. But I really don't imagine this being something alternate SDK implementers want to override.

Reply:
An idea: We might just specify that a context is an opaque object of unspecified type (though we may add a type alias for the real thing). Tracer, Meter, etc. implement (probably in the API layer) methods like `get_current_span_context(context: OpaqueContext = CURRENT_CONTEXT) -> SpanContext` and `set_current_span_context(context: OpaqueContext, newspancontext: SpanContext) -> None`. That way we achieve type safety (kinda).

Reply:
yeah, that's what I think we have to do. 

personally I'd like there to be standards around the setters /getter into context, maybe even via a wrapper object, and to have all SDKs adhere to that. I'm probably limited in imagination here but I don't feel like there's a lot of uses cases where one need to interact with the context differently.
 @c24t @reyang I know you've had thoughts on this in the past. Any opinions on this?

Reply:
Well, I could imagine storing additional custom information in the context that doesn't fit into the predefined Span/Distributed context. E.g., (just a random example from the top of my mind) a handle to some statistics object that counts how often certain OTel operations were called.